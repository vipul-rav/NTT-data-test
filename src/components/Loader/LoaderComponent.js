import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Spinner from 'react-bootstrap/Spinner';

const LoaderComponent = () => (
  <Container>
    <Row>
      <Col>
        <Spinner animation="border" />
      </Col>
    </Row>
  </Container>
);

export { LoaderComponent };
