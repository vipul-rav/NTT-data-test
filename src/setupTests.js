import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

const AllTheProviders = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
