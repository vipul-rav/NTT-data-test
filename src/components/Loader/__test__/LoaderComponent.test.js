import React from 'react';
import { cleanup } from '@testing-library/react';
import { render } from '../../../setupTests';
import { LoaderComponent } from '../LoaderComponent';

const setup = (overridesProps) => {
  const props = {
    ...overridesProps
  };

  const { container, getByText } = render(<LoaderComponent {...props} />);

  return {
    props,
    container,
    getByText
  };
};

describe('LoaderComponent', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
