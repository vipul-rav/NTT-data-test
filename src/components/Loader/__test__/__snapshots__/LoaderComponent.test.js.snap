// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderComponent should render 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="spinner-border"
        />
        Please Wait...
      </div>
    </div>
  </div>
</div>
`;
